pkg/formatted/color.go

```diff
@@ -16,19 +16,74 @@ package formatted
 
 import (
 	"io"
+	"sync"
 
 	"github.com/fatih/color"
 )
 
+type counter struct {
```

Comment:
when I mentione RLock and Lock() I was hinting to having a single mutex in the rainbow. which will be RLocked for checking if the item is in cache and if not, Lock() to add the item in cache. but i like your implementation to create a round robin counter but for the counter itself, `Get` and `increase` isn't quite as useful as just a `next` which returns the next value. 

please consider one of these implemention for counter. 1. that uses atomic, the other that uses a mutex

```
type atomicCounter struct {
  value     uint32
  threshold int
}

func (c *atomicCounter) next() int {
  v := atomic.AddUint32(&c.value, 1)
  next := int(v-1) % c.threshold
  atomic.CompareAndSwapUint32(&c.value, uint32(c.threshold), 0)
  return next

}

type counter struct {
  value     int
  threshold int
  m         sync.Mutex
}

func (c *counter) next() int {
  c.m.Lock()
  defer c.m.Unlock()

  v := c.value
  c.value = (v + 1) % c.threshold
  return v

}
```

and the rainbow would be some what like this 
```
type rainbow struct {
  cache   sync.Map
  counter counter
}

func newRainbow() *rainbow {
  return &rainbow{
    counter: counter{threshold: len(palette)},
  }

}

func (r *rainbow) colorFor(x string) color {
  if value, ok := r.cache.Load(x); ok {
    return value.(color)
  }

  clr := palette[r.counter.next()]
  r.cache.Store(x, clr)
  return clr
}
```

Reply:
I am not really sure why we would want to prematurely optimizing  this on a CLI with a very low requirement for scaling but since you wrote the code i made the change.